components/support/nimbus-fml/src/commands.rs

```diff
@@ -0,0 +1,94 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use anyhow::{bail, Error, Result};
+use std::path::Path;
+use std::path::PathBuf;
+
+pub(crate) struct GenerateStructCmd {
+    pub(crate) manifest: PathBuf,
+    pub(crate) output: PathBuf,
+    pub(crate) language: TargetLanguage,
+    pub(crate) load_from_ir: bool,
+    pub(crate) channel: String,
+}
+
+pub(crate) struct GenerateExperimenterManifestCmd {
+    pub(crate) manifest: PathBuf,
+    pub(crate) output: PathBuf,
+    pub(crate) language: TargetLanguage,
+    pub(crate) load_from_ir: bool,
+    pub(crate) channel: String,
+}
+
+pub(crate) struct GenerateIRCmd {
+    pub(crate) manifest: PathBuf,
+    pub(crate) output: PathBuf,
+    pub(crate) load_from_ir: bool,
+    pub(crate) channel: String,
+}
+
+#[derive(Eq, PartialEq, Hash, Debug)]
+pub enum TargetLanguage {
```

Comment:
nano nit: should we be consistent in visibility between this enum and the structs above? I have no preference since this crate is a binary anyways 🤷  and not worth changing unless there is another commit coming